File: src/main/java/com/zuminX/annotations/swagger/ApiModel.java
package com.zuminX.annotations.swagger;

import com.zuminX.annotations.AnnotationAttr;
import com.zuminX.annotations.AnnotationStr;
import com.zuminX.names.ClassName;
import com.zuminX.names.SwaggerAnnotation;
import java.util.List;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;

/**
 * 对应io.swagger.annotations.ApiModel
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@EqualsAndHashCode(callSuper = true)
public class ApiModel extends AnnotationStr {

  @AnnotationAttr
  private String value;

  @AnnotationAttr(show = true)
  private String description;

  @AnnotationAttr
  private ClassName parent;

  @AnnotationAttr
  private String discriminator;

  @AnnotationAttr
  private List<ClassName> subTypes;

  @AnnotationAttr
  private String reference;

  @Override
  public ClassName getClassName() {
    return SwaggerAnnotation.API_MODEL;
  }
}

File: src/main/java/com/zuminX/annotations/swagger/ApiModel.kt
package com.zuminX.annotations.swagger

import com.zuminX.annotations.AnnotationAttr
import com.zuminX.annotations.AnnotationStr
import com.zuminX.names.ClassName
import com.zuminX.names.annotation.API_MODEL

/**
 * 对应io.swagger.annotations.ApiModel
 */
class ApiModel : AnnotationStr() {

  @field:AnnotationAttr
  var value: String? = null

  @field:AnnotationAttr(show = true)
  var description: String? = null

  @field:AnnotationAttr
  var parent: ClassName? = null

  @field:AnnotationAttr
  var discriminator: String? = null

  @field:AnnotationAttr
  var subTypes: List<ClassName>? = null

  @field:AnnotationAttr
  var reference: String? = null

  override fun getClassName(): ClassName = API_MODEL
}